buildenv/jenkins/common/build.groovy

```diff
@@ -529,10 +539,38 @@ def archive_diagnostics() {
             }
         }
         // Note: to preserve the files ACLs set _OS390_USTAR=Y env variable (see variable files)
-        sh "find . -name 'core.*.dmp' -o -name 'javacore.*.txt' -o -name 'Snap.*.trc' -o -name 'jitdump.*.dmp' | sed 's#^./##' | pax -wzf ${diagnosticsFilename}"
+        archiveCmd = "pax -wzf ${diagnosticsFilename}"
     } else {
-        sh "find . -name 'core.*.dmp' -o -name 'javacore.*.txt' -o -name 'Snap.*.trc' -o -name 'jitdump.*.dmp' -o -name ddr_info -o -name j9ddr.dat -o -name superset.dat -o -name '*.annt.h' -o -name '*.stub.h' -o -name '*.dSYM' -o -path '*/make-support/failure-logs/*' | sed 's#^./##' | tar -zcvf ${diagnosticsFilename} -T -"
+        findCmd += " -o -name ddr_info"
+        findCmd += " -o -name j9ddr.dat"
+        findCmd += " -o -name superset.dat"
+        findCmd += " -o -name '*.annt.h'"
+        findCmd += " -o -name '*.stub.h'"
+        findCmd += " -o -path '*/make-support/failure-logs/*'"
     }
+
+    // If a core was generated, then include all of the binaries and any
+    // separate debug info in the diagnostics archive to help debug the core.
+    if (sh(returnStatus: true, script: "find . ${findNameCore} | grep -q .") == 0) {
+        // JDK bin directory. On Windows this will contain .exe and .dll files,
+        // but on most systems the binaries in it won't have any file extension
+        // we could use to identify them, so just include everything in there.
+        findCmd += " -o -path '*/build/*/jdk/bin/*'"
```

Comment:
I'd collect everything here no matter if the core is found or not. Maybe they can be used to recreate the failure manually.

Reply:
Ah, I was just trying to avoid collecting everything when we have the usual kind of build failure, i.e. an error where the build compiler complains about the source, or javac complains about some Java source. Failures like that should be possible to recreate manually just based on the source.

If you want to collect this all even in those cases, I can easily make that change.

Reply:
It's a good idea. I was thinking about the case where the core file didn't work. How about collecting the info if any core, javacore, Snap, jitdump is present.

Reply:
Now collecting this type of stuff if any JVM crash data is found